addons/project/models/project_project.py

```diff
@@ -416,6 +416,20 @@ def copy(self, default=None):
                 old_project.map_tasks(new_project.id)
             if not self.active:
                 new_project.with_context(active_test=False).tasks.active = True
+            # Copy the shared embedded actions in the new project
+            shared_embedded_actions = self.env['ir.embedded.actions'].search([
+                ('parent_res_id', '=', old_project.id),
+                ('parent_res_model', '=', 'project.project'),
+                ('user_id', '=', False),
+            ])
+            copy_shared_embedded_actions = shared_embedded_actions.copy({'parent_res_id': new_project.id})
+            for i in range(len(shared_embedded_actions)):
```

Comment:
Use `zip` and destructuring
```suggestion
            for original_action, copied_action in zip(shared_embedded_actions, copy_shared_embedded_actions):
```